File: primos/teste.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading;

class Program
{
    private static List<double> primes = new List<double>();

    static void Main(string[] args)
    {
        double max = Math.Pow(10, 6);
        EnginePreviousPrimes(max);
    }

    public static void EnginePreviousPrimes(double max)
    {
        Console.WriteLine("Max: " + max);
        primes.Add(2.0);
        
        double raiz;
        double n = 3;
        bool isPrime;
        Stopwatch stopwatch = Stopwatch.StartNew();  // Início do contador de tempo

        while (n <= max)
        {
            isPrime = true;
            raiz = Math.Sqrt(n) + 1;
            
            foreach (double k in primes)
            {
                if (n % k == 0)
                {
                    isPrime = false;
                    break;
                }
                if (k > raiz) break;
            }
            
            if (isPrime)
                primes.Add(n);
            
            n += 2.0;
        }

        Show();
        stopwatch.Stop();
        Console.WriteLine($"Last : {n}");
        Console.WriteLine($"Tempo (s): {stopwatch.Elapsed.TotalSeconds:F9}");
    }

    public static void Sleep(double seconds)
    {
        try
        {
            Thread.Sleep((int)(seconds * 1000));
        }
        catch (Exception e)
        {
            Console.WriteLine("Erro ao pausar o programa: " + e.Message);
        }
    }

    public static void Show()
    {
        int qt = 0;
        foreach (double prime in primes)
        {
            Console.WriteLine(prime);
            qt++;
        }
        Console.WriteLine("Quantidade : " + qt);
    }
}

File: primos/teste.py
import time
import math

# Lista para armazenar os números primos
primes = [2.0]

def main():
    max_val = 10**7
    engine_previous_primes(max_val)

def engine_previous_primes(max_val):
    print(max_val)
    raiz = 1
    n = 3
    init = time.time()  # Início do contador de tempo

    while n <= max_val:
        p = True
        raiz = math.sqrt(n) + 1
        for k in primes:
            if n % k == 0:
                p = False
                break
            if k > raiz:
                break
        if p:
            primes.append(n)
        n += 2.0

    show()
    print(f"Last : {n}")
    print(f"Tempo (s): {time.time() - init:.9f}")

def sleep(tempo):
    try:
        time.sleep(tempo)
    except Exception as e:
        pass

def show():
    qt = 0
    for n in primes:
        print(n)
        qt += 1
    print(f"Quantidade : {qt}")

# Executa o programa principal
if __name__ == "__main__":
    main()
